### DISCUSSION TITLE
Sidebar Vuejs component localilzation

### DISCUSSION THREAD

**Discussion by joeyramone76** [area17/twill#1497] [Q&A]
Hi to all and thanks to the developers team for the great work they have done!
I use Twill for several projects but I have not been able to localize "Expired" strings and dates (in italian) when using publish_start_date and publish_end_date.  
Is there a way to translate the label "expired" and set the it-IT locale format for  the publish_start_date and publish_end_date fields in the right column (see attachment) in the admin console without having to change the core (the vuejs components in this case)?

Thanks!

<img width="302" alt="Label and date to translate/localize" src="https://user-images.githubusercontent.com/1647785/157126904-2df26215-beb7-4bd9-baba-41bf6982b530.png">

**Comment by haringsrob** [comment]
Hey @joeyramone76 ,

This string should already be translatable,  but it might not yet be in the language files.

The key is `publisher.expired`.

You can fix this by either adding the translation in your codebase or by contributing it back.

To add it in your codebase:

<img width="586" alt="Screenshot 2022-03-08 at 08 12 32" src="https://user-images.githubusercontent.com/866743/157185598-d7e7375d-f6ff-457c-9fae-6ebae35ad8c7.png">

To contribute it back you can change the file here using English as the reference of where to put it.

https://github.com/area17/twill/tree/2.x/lang

**Reply by joeyramone76** [reply]
Hi @haringsrob,
 many thanks for the reply but this solution don't works for me.
I've adding the key in resources/lang/vendor/twill/it/lang (see attachment), run artisan twill:build --noInstall, artisan view:clear and artisan config:clear but the string remain "Expired" in the admin console. 
I found this strings under vendor/area17/twill/frontend/js in .vue and js files (see attachments); where am I wrong? 

For the date localization (I pointed out the problem in the first comment) have you a solution?

------------------------------
<img width="228" alt="Schermata 2022-03-08 alle 10 09 43" src="https://user-images.githubusercontent.com/1647785/157207376-f2e6dfc3-7c42-4ecb-b283-2ef0e302c581.png">
<img width="509" alt="Schermata 2022-03-08 alle 10 09 30" src="https://user-images.githubusercontent.com/1647785/157207378-9953e66a-840f-470e-ad36-0944124f9449.png">

<img width="1345" alt="Schermata 2022-03-08 alle 10 08 14" src="https://user-images.githubusercontent.com/1647785/157207382-ec62cc88-b375-4ee6-a70a-a640d221b734.png">

<img width="828" alt="Schermata 2022-03-08 alle 10 22 20" src="https://user-images.githubusercontent.com/1647785/157207361-dfdbfdbe-989d-4358-9b2d-7a8f8c2234e3.png">
<img width="872" alt="Schermata 2022-03-08 alle 10 22 13" src="https://user-images.githubusercontent.com/1647785/157207371-b523a710-842c-4374-aed2-3fcc0ec6a703.png">
<img width="791" alt="Schermata 2022-03-08 alle 10 22 00" src="https://user-images.githubusercontent.com/1647785/157207374-65027515-11ac-4293-a91c-98355e4bb578.png">
